compiler/rustc_mir_build/src/builder/scope.rs

```diff
@@ -779,14 +790,96 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
-                let scope = &self.scopes.const_continuable_scopes[break_index];
+                let scope_index = self.scopes.scope_index(
+                    self.scopes.const_continuable_scopes[break_index].region_scope,
+                    span,
+                );
+                let scope = &mut self.scopes.const_continuable_scopes[break_index];
                 self.cfg.push_assign(block, source_info, discr, rvalue);
+                let drop_and_continue_block = self.cfg.start_new_block();
+                let imaginary_target = self.cfg.start_new_block();
                 self.cfg.terminate(
                     block,
                     source_info,
-                    TerminatorKind::FalseEdge { real_target, imaginary_target: scope.loop_head },
+                    TerminatorKind::FalseEdge {
+                        real_target: drop_and_continue_block,
+                        imaginary_target,
+                    },
                 );
 
+                let drops = &mut scope.break_drops;
+
+                let drop_idx = self.scopes.scopes[scope_index + 1..]
+                    .iter()
+                    .flat_map(|scope| &scope.drops)
+                    .fold(ROOT_NODE, |drop_idx, &drop| drops.add_drop(drop, drop_idx));
+
+                drops.add_entry_point(imaginary_target, drop_idx);
+
+                self.cfg.terminate(imaginary_target, source_info, TerminatorKind::UnwindResume);
+
+                // FIXME add to drop tree for loop_head
+
+                let region_scope = scope.region_scope;
+                let scope_index = self.scopes.scope_index(region_scope, span);
+                let mut drops = DropTree::new();
+
+                let drop_idx = self.scopes.scopes[scope_index + 1..]
+                    .iter()
+                    .flat_map(|scope| &scope.drops)
+                    .fold(ROOT_NODE, |drop_idx, &drop| drops.add_drop(drop, drop_idx));
+
+                drops.add_entry_point(drop_and_continue_block, drop_idx);
+
+                // `build_drop_trees` doesn't have access to our source_info, so we
+                // create a dummy terminator now. `TerminatorKind::UnwindResume` is used
+                // because MIR type checking will panic if it hasn't been overwritten.
+                // (See `<ExitScopes as DropTreeBuilder>::link_entry_point`.)
+                self.cfg.terminate(
+                    drop_and_continue_block,
+                    source_info,
+                    TerminatorKind::UnwindResume,
+                );
+
+                {
+                    let this = &mut *self;
+                    let blocks = drops.build_mir::<ExitScopes>(&mut this.cfg, Some(real_target));
+                    //let is_coroutine = this.coroutine.is_some();
+
+                    /*// Link the exit drop tree to unwind drop tree.
+                    if drops.drops.iter().any(|drop_node| drop_node.data.kind == DropKind::Value) {
```

Comment:
this is just WIP I assume?

Reply:
Yes, I copied it from `build_exit_tree`. Will need to be enabled at some point.